lib/dao.js

```diff
@@ -1196,7 +1196,8 @@ DataAccessObject._coerce = function (where) {
       }
     } else {
       if (val != null) {
-        if (operator === 'regexp' && val instanceof RegExp) {
+        if (val instanceof RegExp ||
```

Comment:
You should check `(operator === null && val instanceof RegExp)`. If it's true, set the operator to `regexp`. This way, `{name: /A/}` will be normalized as `{name: {regexp: /A/}}` and you don't have to change any code in connectors.